src/Engine/ProtoCore/Reflection/Mirror.cs

```diff
@@ -337,23 +354,30 @@ internal ClassMirror(StackValue svData, ProtoCore.Core core)
             /// <summary>
             /// Returns the constructors and static methods and properties 
             /// belonging to the type and its base types
+            /// Excludes hidden methods and properties from base types.
             /// </summary>
             /// <returns></returns>
             public IEnumerable<StaticMirror> GetMembers()
             {
                 // TODO: Factor out reflection functionality for both LibraryServices and Mirrors
                 List<StaticMirror> members = new List<StaticMirror>();
+                members.AddRange(GetConstructors().GroupBy(x => x.Name).Select(y => y.First()));
+                members.AddRange(GetFunctions().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
+                members.AddRange(GetProperties().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
+
+                // Filter out hidden functions/properties:
+                // Create a set of unique function, property and constructor descriptions.
+                // In this case we use ToString() to get the uniques description of the members (func signature, property names, constructor names).
+                var derivedClassMembers = new HashSet<string>();
+                members.ForEach(x => derivedClassMembers.Add(x.ToString()));
 
                 IEnumerable<ClassMirror> baseClasses = this.GetClassHierarchy();
                 foreach (var baseClass in baseClasses)
                 {
-                    members.AddRange(baseClass.GetFunctions().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
-                    members.AddRange(baseClass.GetProperties().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
+                    members.AddRange(baseClass.GetFunctions().Where(m => m.IsStatic && !derivedClassMembers.Contains(m.ToString())).GroupBy(x => x.Name).Select(y => y.First()));
+                    members.AddRange(baseClass.GetProperties().Where(m => m.IsStatic && !derivedClassMembers.Contains(m.ToString())).GroupBy(x => x.Name).Select(y => y.First()));
                 }
 
-                members.AddRange(this.GetConstructors().GroupBy(x => x.Name).Select(y => y.First()));
-                members.AddRange(this.GetFunctions().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
-                members.AddRange(this.GetProperties().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
                 return members;
```

Comment:
it seems I also changed the order of the returned members here.
It used to be base classes first then the current class.

With my changes it is reversed.
Seems there was a unit test that covered the order returned by `GetCompletionsOnType` which calls GetMembers.
Should we just sort them alphabetically ?

Opinions @mjkkirschner @aparajit-pratap ?

Reply:
It would be good to arrange them by constructors followed by methods, followed by properties. That way it will also be sort of aligning with node autocomplete results.

Reply:
Ok, sounds good to me. That is the way it is done in the current state of the PR.